crates/hir-expand/src/builtin_fn_macro.rs

```diff
@@ -671,7 +671,7 @@ fn option_env_expand(
 
     let expanded = match get_env_inner(db, arg_id, &key) {
         None => quote! { #DOLLAR_CRATE::option::Option::None::<&str> },
-        Some(s) => quote! { #DOLLAR_CRATE::option::Some(#s) },
+        Some(s) => quote! { #DOLLAR_CRATE::option::Option::Some(#s) },
```

Comment:
@Veykril are you sure `::core::option::Option` wouldn't work better here? I think we fail to expand a couple of these in `analysis-stats`.

Reply:
> and is $crate really correct?

The `option_env!` macro is defined in core, so I would expect it to be.

Reply:
I guess we're not handling it correctly, then.

Reply:
Odd, we should be handling `$crate` for these properly 🤔

Reply:
Yeah, we don't resolve it (see the three instances in `version.rs`), while it works with `::core::option::Option`.

Reply:
Maybe, I mean I am looking into rewriting the token map stuff (though I don't have a lot of time to put into r-a currently), and I imagine this will also require me to change how we handle `$crate` as this involves rewriting/adding hygiene proper handling.